awswrangler/s3/_read_orc.py

```diff
@@ -0,0 +1,582 @@
+"""Amazon S3 Read ORC Module (PRIVATE)."""
+
+import datetime
+import itertools
+import logging
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+import boto3
+import pandas as pd
+import pyarrow as pa
+import pyarrow.dataset
+import pyarrow.orc
+from typing_extensions import Literal
+
+from awswrangler import _data_types, _utils, exceptions
+from awswrangler._arrow import _add_table_partitions
+from awswrangler._config import apply_configs
+from awswrangler._distributed import engine
+from awswrangler._executor import _BaseExecutor, _get_executor
+from awswrangler.s3._fs import open_s3_object
+from awswrangler.s3._list import _path2list
+from awswrangler.s3._read import (
+    _apply_partition_filter,
+    _check_version_id,
+    _extract_partitions_dtypes_from_table_details,
+    _get_path_ignore_suffix,
+    _get_path_root,
+    _get_paths_for_glue_table,
+    _TableMetadataReader,
+)
+from awswrangler.typing import RaySettings, _ReadTableMetadataReturnValue
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3 import S3Client
+
+FULL_READ_S3_BLOCK_SIZE = 20_971_520  # 20 MB (20 * 2**20)
+METADATA_READ_S3_BLOCK_SIZE = 131_072  # 128 KB (128 * 2**10)
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _pyarrow_orc_file_wrapper(source: Any) -> pyarrow.orc.ORCFile:
+    try:
+        return pyarrow.orc.ORCFile(source=source)
+    except pyarrow.ArrowInvalid as ex:
+        if str(ex) == "ORC file size is 0 bytes":
+            _logger.warning("Ignoring empty file...")
+            return None
+        raise
+
+
+@engine.dispatch_on_engine
+def _read_orc_metadata_file(
```

Comment:
Is the reason for keeping this function instead of only having the reader implementation in the engine dispatch logic?